### PR TITLE
[dv/otp_ctrl] OTP reaches V2

### PR DESCRIPTION
**Pull request by cindychip**
As discussed in the meeting, we plan to wavie prim_tl_o/i until the
reggen tool is optmized.

From the previous V2 status, here are the design changes and what we
have covered:
- prim_tl_i/o: add basic support. But not a complete support because we expect some reggen related change.
  - DV open source currently supports read/write only.
    Merged PR: https://github.com/lowRISC/opentitan/pull/10234
- Vendor test partition
  - DV change: treat exactly same as other sw partition except uncorrectable ECC does not trigger fatal alerts
  Merged PR: https://github.com/lowRISC/opentitan/pull/10268
- Output Cio_test_o, cio_test_en_o
  - DV changes: Assertion checks
    https://github.com/lowRISC/opentitan/pull/10229/files
- lc_otp_vendor_test_i/o
  - DV changes: Assertion checks
    https://github.com/lowRISC/opentitan/pull/10229/files
- UNR updates:
  Added a partition (vendor test partition, so all previous partition related UNR is misaligned)
  Vendor related I/Os are tied to 0.
  V2S related assertions not covered and not detected by UNR.
  UNRs that needs to be reviewed again are filed here:  https://github.com/lowRISC/opentitan/issues/10313

Above items have been reviewed with designer Michael and there is one
pending issue: https://github.com/lowRISC/opentitan/pull/10317 but we
believe it does not affect our V2 progress because it is a small
alignment.

Regression status: https://reports.opentitan.org/hw/ip/otp_ctrl/dv/latest/results.html 
image

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

File: hw/ip/otp_ctrl/data/otp_ctrl.prj.hjson
Comment (msfschaffner) [inline hw/ip/otp_ctrl/data/otp_ctrl.prj.hjson:27]: Is this maybe a comment that should go into the V2 checklist instead?

**Review by msfschaffner — COMMENTED** [review]
LGTM, modulo the comment. 
I leave it up to DV folks to approve.

**Comment by cindychip** [comment]
That is a good point! Thanks so much for pointing it out.

**Comment by sriyerg** [comment]
Seems like you have applied UNR generated exclusions as well. There is still an open question regarding UNR excluding counter measures inadvertently because that is redundant logic that cannot be exercised without forces. I dont mean to increase your workload, but could you just regenerate the coverage report WITHOUT the UNR generated exclusions to check if we are still over V2 threshold?

**Comment by cindychip** [comment]
Hi Sri,

No worries, and thanks for your question, it is definitely a valid concern.
Without the UNR it won't pass the V2 threshold.
One main reason is that all the buffer/unbuffered partitions are
implemented with the same RTL template with different configurations. It is
not possible to reach 90% if we do not apply certain exclusions.
On the other hand, Michael and I reviewed the UNR and filed an issue to
track all the security related exclusions:
https://github.com/lowRISC/opentitan/issues/10313
We plan to revisit this UNR exclusion in V2S.
Please let me know if you have any concerns.

Cindy

  Cindy Chen

**Review by weicaiyang — APPROVED** [review]
LGTM

**Comment by sriyerg** [comment]
Could we regenerate UNR exclusions with the standardized countermeasure modules black boxed?

**Comment by cindychip** [comment]
sure! I can do that.

  Cindy Chen

**Comment by cindychip** [comment]
gated by recent issue: https://github.com/lowRISC/opentitan/issues/10478

**Comment by cindychip** [comment]
PR https://github.com/lowRISC/opentitan/pull/10614 is newly generated UNR file that blackbox the countermeasure, and is reviewed and merged. The regression coverage with latest exclusion should show up on tonight's regression.

**Comment by cindychip** [comment]
@sriyerg OTP V2 is ready for review again ;) Thanks for the suggestions. I removed the countermeasure related UNRs (via blackboxing), and was reviewed and merged on Friday. So the current regression results reflects the coverage with the new UNRs.

**Review by sriyerg — APPROVED** [review]
LGTM from me. 
I will leave it to @msfschaffner for final approval - I will assume that @msfschaffner has validated the UNR generated exclusions.

**Comment by msfschaffner** [comment]
Yes, we've reviewed the UNR-generated exclusions.

**Comment by cindychip** [comment]
Thanks everyone.

**Pull request merged**
